### DISCUSSION TITLE
Help with expanding relations

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#4440] [Q&A] [answered]
Hi, I have the following tables (with relations to each other):

![image](https://github.com/pocketbase/pocketbase/assets/12494537/07bf8720-718f-44a4-976c-a5775f299027)
![image](https://github.com/pocketbase/pocketbase/assets/12494537/c4e4f5a2-b4e6-4461-8067-50250ea531c9)

I am trying to get this query working:
`locals.userPb.collection('Class').getFullList(50, { sort: '-created', expand: 'SubjectOffering(class).subjectOffering' })`

But all I get is this:
`[
  {
    collectionId: 'n8vqidrr972omwx',
    collectionName: 'Class',
    created: '2024-02-27 03:54:11.039Z',
    endDate: '',
    examDate: '2024-02-08 00:00:00.000Z',
    id: '7zau7rat6mckv4b',
    name: 'IGCSE-AddMaths-W-25-1',
    slotDay: 'Monday',
    slotDuration: '2',
    slotTime: '12:45',
    startDate: '2024-02-08 00:00:00.000Z',
    subjectOffering: 'wcqxpd6kf8x79o9',
    type: 'group',
    updated: '2024-02-27 04:49:31.941Z',
    expand: {}
  }
]`

My problem is the expand field is not populated with the 'subjectOffering' details. I can't seem to figure out what I am doing wrong. Any help would be greatly appreciated.

**Comment by ganigeorgiev** [comment, marked as answer]
Back-relation expand is only for the cases when the relation field is not in the main/targetted collection.

That's not the case based on your screenshots because you have a direct relation from "Class" to "SubjectOffering". Or in other words, try with:

```js
locals.userPb.collection('Class').getFullList({
    sort: '-created',
    expand: 'subjectOffering'
})
```

If it doesn't work, then make sure that your client request satisfy the View API rule of the "SubjectOffering" collection.

**Reply by rameshkumar70** [reply]
I'm still trying to wrap my head around this expanding of records and the back-relation concepts. I tried your recommendation and it works perfectly. Thank you so much. 

I still would like to understand though the concept underlying the expansion using syntax like 'A(B).C' ;  what do A, B, C stand for? Is A a collection name, B a field name, C a field name?

**Reply by ganigeorgiev** [reply]
@rameshkumar70 I don't think I can explain it better than what is already in the documentation - https://pocketbase.io/docs/working-with-relations/#back-relations (note that the parenthesis syntax was soft-deprecated in v0.22.0; see also https://github.com/pocketbase/pocketbase/discussions/4417).

**Reply by rameshkumar70** [reply]
Thank you so much.
